ard_controller/ard_controller.ino

```diff
@@ -245,20 +287,89 @@ void rx_processor() {
   }
 }
 
+// reads current position estimate from odrive and stores in global array
+void update_cur_pos() {
+  odrv_leftleg_ser.write("r axis0.encoder.pos_estimate\n"); // left hip
+  cur_joint_pos[0] = constrain(fmod(odrv_leftleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_leftleg_ser.write("r axis1.encoder.pos_estimate\n"); // left knee
+  cur_joint_pos[1] = constrain(fmod(odrv_leftleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightleg_ser.write("r axis0.encoder.pos_estimate\n"); // right hip
+  cur_joint_pos[2] = constrain(fmod(odrv_rightleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightleg_ser.write("r axis1.encoder.pos_estimate\n"); // right knee
+  cur_joint_pos[3] = constrain(fmod(odrv_rightleg.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+
+  odrv_leftarm_ser.write("r axis0.encoder.pos_estimate\n"); // left shoulder
+  cur_joint_pos[4] = constrain(fmod(odrv_leftarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_leftarm_ser.write("r axis1.encoder.pos_estimate\n"); // left elbow
+  cur_joint_pos[5] = constrain(fmod(odrv_leftarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightarm_ser.write("r axis0.encoder.pos_estimate\n"); // right shoulder
+  cur_joint_pos[6] = constrain(fmod(odrv_rightarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+  odrv_rightarm_ser.write("r axis1.encoder.pos_estimate\n"); // right elbow
+  cur_joint_pos[7] = constrain(fmod(odrv_rightarm.readFloat()*(360/GEAR_RATIO), 360), 0, 360);
+}
```

Comment:
Seems like a lot of repeat code and some room for modularity (using a loop?). With that being said, I'm not sure how easy it would be to implement in Arduino cause I know that C can be a bitch. Can you think of any quick, easy way to resolve this @mahajanrevant?  Otherwise, it may not be pretty, but I think we might just have to keep it.

Reply:
yea as I think about it, the `odrv_*` would ideally be a dictionary or a richer data structure, but I don't think that Arudino has that easy available.

Reply:
@agupta231  the only way I can think is using std:map library which is just a dictionary for c++. Not sure we want to do that right now given the objectives we are trying to achieve. It ain't the prettiest but gets the job done

Reply:
Yea the `odrv_*` bits as @agupta231 mentioned are what make it hard to iterate these statements. There are probably ways to fix it but as @mahajanrevant mentioned, I think we should just accept the ugliness for now in the context of moving forward.